### PR TITLE
More misc billing

### PR DESCRIPTION
**Pull request by stephenwaite**
improved functioning

### CODE REVIEW

File: sql/database.sql [hunk 1]
```diff
@@ -1387,6 +1387,7 @@ CREATE TABLE `form_misc_billing_options` (
   `outside_lab` tinyint(1) default NULL,
   `lab_amount` decimal(5,2) default NULL,
   `is_unable_to_work` tinyint(1) default NULL,
+  `onset_date` date defaults NULL,
```
Comment (bradymiller) [inline sql/database.sql]: typo, should be default

File: interface/forms/misc_billing_options/new.php [hunk 2]
```diff
@@ -2,13 +2,12 @@
...
- * Copyright (C) 2007 Bo Huynh
- * Copyright (C) 2016 Terry Hill <terry@lillysystems.com>
- * Copyright (C) 2017 Brady Miller <brady.g.miller@gmail.com>
- *
  * @package OpenEMR
  * @author Terry Hill <terry@lilysystems.com>
  * @author Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (C) 2007 Bo Huynh
+ * @copyright Copyright (C) 2016 Terry Hill <terry@lillysystems.com>
+ * @copyright Copyright (C) 2017 Brady Miller <brady.g.miller@gmail.com>
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php]: Don't forget to add yourself to these headers. Pretty much any modification allows you to add yourself as author/copyright.

File: interface/forms/misc_billing_options/new.php [hunk 3]
```diff
@@ -22,14 +21,17 @@
...
 
-$mboquery = array();
-$mboquery = sqlquery("SELECT fmbo.id FROM form_misc_billing_options AS fmbo 
-                      INNER JOIN forms ON (fmbo.id = forms.form_id) WHERE 
-                      forms.encounter = ?", array($encounter));
-if ($mboquery) {
-  $formid = 0 + $mboquery['id'];
-} else {
-  $formid   = 0 + formData('id', 'G');
+$formid   = 0 + formData('id', 'G');
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php]: this appears to be a duplicate line, so can remove

File: interface/forms/misc_billing_options/new.php [hunk 4]
```diff
@@ -23,6 +22,18 @@
 }
 
 $formid   = 0 + formData('id', 'G');
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php]: above is a duplicate line, so can remove it

Comment (stephenwaite) [inline interface/forms/misc_billing_options/new.php]: thanks for the quick review

**Comment by bradymiller** [comment]
I'm gonna do a quick test and then bring this into the codebase. Is the billing part testing well for you?

**Comment by bradymiller** [comment]
Very nice code!
Did sanity testing and looks good (misc form looks good, saves well, only allows 1 form per encounter, and produces x12 and hfca; note I can't really test x12 and hfca but no errors when producing them).
Going to being this into the codebase. Thanks for the nice contribution!
-brady

**Pull request merged**